### ISSUE TITLE
DiceDB doesn't build on linux/arm64

### ISSUE THREAD

**Issue by asutosh97** [DiceDB/dice#58]
**Describe the bug**
DiceDB doesn't build on `linux/arm64`. It gives the following error:

```bash
~:/dice# go build
# github.com/dicedb/dice/core
core/eval.go:178:44: undefined: syscall.SYS_FORK
```

As mentioned, it is because of the use of `syscall.SYS_FORK` in `eval.go`

```go
func evalBGREWRITEAOF(args []string) []byte {
	newChild, _, _ := syscall.Syscall(syscall.SYS_FORK, 0, 0, 0)
        ...
        ...
}
```

**To Reproduce**
Run `go build` on a `linux/arm64` machine

**Expected behavior**
The build should happen successfully without any error messages and generate a binary

**Additional context**
Upon simply googling about this, came across this issue in golang repo: https://github.com/golang/go/issues/11981
Also, found a related stackoverflow answer: https://stackoverflow.com/questions/28370646/how-do-i-fork-a-go-process/28371586#28371586

TL;DR: Calling `fork()` using `Syscall` is unsafe, recommended API is `ForkExec` instead

**Comment by rohanverma94** [comment]
Actually, the problem is not related to FORK but it is POSIX. When I wrote this, I was clear that FORK is not portable across platforms, on POSIX systems maybe. On windows port of dice for example, I've to implement CoW pages using kernel API to mimic FORK call. I can write a FIX for arm64 if it is the target you want, that is very straight forward. 

The issue with lot of syscalls like fork, pipe etc in arm64 is that arm64/linux doesn't implements address isolations perfectly. 

Regarding safety of FORK, the ForKExec is indeed a combination of Fork and Exec, though for the usecases we want the plain FORK is suitable as it gives granularity. If @arpitbbhayani approves the target aarch64 or arm64 linux then I can issue the fix right away.

**Comment by rohanverma94** [comment]
I'll issue a fix for arm64 (m1 compatible) and the entire AOF stub will be upgraded to a stable implementation tomorrow.

**Comment by JyotinderSingh** [comment]
Completed in #92
